craft_parts/plugins/qmake_plugin.py

```diff
@@ -0,0 +1,145 @@
+        :raise pydantic.ValidationError: If validation fails.
+        """
+        plugin_data = extract_plugin_properties(
+            data, plugin_name="qmake", required=["source"]
+        )
+        return cls(**plugin_data)
+
+
+class QmakePlugin(Plugin):
+    """The qmake plugin is useful for building qmake-based parts.
+
+    These are projects that are built using .pro files.
+
+    This plugin uses the common plugin keywords as well as those for "sources".
+    For more information check the 'plugins' topic for the former and the
+    'sources' topic for the latter.
+
+    Additionally, this plugin uses the following plugin-specific keywords:
+
+    - qmake-parameters:
+      (list of strings)
+      additional options to pass to the qmake invocation.
+
+    - qmake-project-file:
+      (string)
+      the qmake project file to use. This is usually only needed if
+      qmake can not determine what project file to use on its own.
+
+    - qmake_major_version:
+      (int)
+      set the qt major version. This is only needed to support qt6 builds.
+      Version 5 is default.
+    """
+
+    properties_class = QmakePluginProperties
+
+    def get_build_snaps(self) -> Set[str]:
+        """Return a set of required snaps to install in the build environment."""
+        return set()
+
+    def get_build_packages(self) -> Set[str]:
+        """Return a set of required packages to install in the build environment."""
+        options = cast(QmakePluginProperties, self._options)
+
+        if options.qmake_major_version == 6:
+            build_packages = {"g++", "make", "qmake6"}
+        else:
+            build_packages = {"g++", "make", "qt5-qmake"}
+        return build_packages
+
+    def get_build_environment(self) -> Dict[str, str]:
+        """Return a dictionary with the environment to use in the build step."""
+        options = cast(QmakePluginProperties, self._options)
+
+        if options.qmake_major_version == 6:
+            return {"QT_SELECT": "qt6"}
+
+        return {"QT_SELECT": "qt5"}
+
+    def get_build_commands(self) -> List[str]:
+        """Return a list of commands to run during the build step."""
+        options = cast(QmakePluginProperties, self._options)
+
+        if options.qmake_major_version == 6:
+            qmake_configure_command = [
+                "qmake6",
+                'QMAKE_CFLAGS+="${CFLAGS:-}"',
+                'QMAKE_CXXFLAGS+="${CXXFLAGS:-}"',
+                'QMAKE_LFLAGS+="${LDFLAGS:-}"',
+            ] + options.qmake_parameters
+        else:
+            qmake_configure_command = [
+                "qmake",
+                'QMAKE_CFLAGS+="${CFLAGS:-}"',
+                'QMAKE_CXXFLAGS+="${CXXFLAGS:-}"',
+                'QMAKE_LFLAGS+="${LDFLAGS:-}"',
+            ] + options.qmake_parameters
+
+        if options.qmake_project_file:
+            qmake_configure_command.extend(
+                [
+                    f'"{self._part_info.part_src_dir}"',
+                    "/",
```

Comment:
Just curious, what does this command-line parameter mean?

Reply:
Sorry, not clear what you are asking here.  I have an optional qmake_project_file: str = "" property so that one can define a non standard .pro file as demonstrated in the test 


    def test_get_build_commands_qmake_project_file(self, setup_method_fixture, new_dir):
        plugin = setup_method_fixture(
            new_dir, properties={"qmake_project_file": "hello.pro"}

Reply:
I asked because this list seems to evaluate to three different command-line arguments (src dir, /, and the project file) and I was wondering what the / could mean here. Or should they assemble a single path such as `srcdir/hello.pro`?

Reply:
Correct, assemble a single path

Reply:
But then, when joining with a space in line 137 below, wouldn't that result in three different arguments?

Reply:
One way to address this could be something like:
```
qmake_configure_command.append(str(self._part_info.part_src_dir / options.qmake_project_file))
```
in line 126 so that this specific entry would be a single argument.

Reply:
Thank you, a big problem is I can't seem to run make tests locally, ruff tests fail with things way beyond my level of knowledge. Is there a way to only run unit tests?

Reply:
You can run `make test-units` or `make test-integrations` to run all of each kind or `pytest -vv tests/some_file.py` to run a single test.